sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -406,6 +406,13 @@ private void initializeGatewayConfigurationReader() {
         // https://msdata.visualstudio.com/CosmosDB/_workitems/edit/332589
     }
 
+    private void updateGatewayProxy() {
+        ((RxGatewayStoreModel)this.gatewayProxy).setGatewayServiceConfigurationReader(this.gatewayConfigurationReader);
+        ((RxGatewayStoreModel)this.gatewayProxy).setCollectionCache(this.collectionCache);
+        ((RxGatewayStoreModel)this.gatewayProxy).setPartitionKeyRangeCache(this.partitionKeyRangeCache);
```

Comment:
these seem to be required for GatewayProxyModel? in that case we should be passing them in the constructor.

why are we using setter?

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/25555#discussion_r754658494